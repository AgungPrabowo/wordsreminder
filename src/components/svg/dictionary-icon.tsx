import React from 'react'
import Svg, { Path } from 'react-native-svg'
import { ThemeContext } from '@contexts/theme-context'

type Props = {
  opacity?: number
  testID?: string
}

const DictionaryIcon = ({ opacity = 1, testID }: Props) => (
  <ThemeContext.Consumer>
    {({ theme }) => (
      <Svg height={28} width={28} viewBox="0 0 24 24" opacity={opacity} testID={testID}>
        <Path d="M0 0h24v24H0z" fill="none" />
        <Path
          fill={theme.primary025}
          d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
        />
      </Svg>
    )}
  </ThemeContext.Consumer>
)

export { DictionaryIcon }
