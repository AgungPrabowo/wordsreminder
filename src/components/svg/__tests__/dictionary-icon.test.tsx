import React from 'react'
import renderer from 'react-test-renderer'
import { DictionaryIcon } from '../dictionary-icon'

describe('DictionaryIcon', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<DictionaryIcon opacity={0.3} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
