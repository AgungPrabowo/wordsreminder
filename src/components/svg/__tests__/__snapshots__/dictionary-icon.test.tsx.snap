// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictionaryIcon should match snapshot 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={28}
  bbWidth={28}
  fill="#fff"
  focusable={false}
  height={28}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      undefined,
      Object {
        "opacity": 1,
      },
      Object {
        "flex": 0,
        "height": 28,
        "width": 28,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={28}
>
  <RNSVGGroup
    fill={4294967295}
    fillOpacity={1}
    fillRule={1}
    font={Object {}}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "fill",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={null}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth={1}
    vectorEffect={0}
    x={0}
    y={0}
  >
    <RNSVGPath
      d="M0 0h24v24H0z"
      fill={null}
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
    <RNSVGPath
      d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
      fill={4278190080}
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={Array []}
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
